code/midtown/mmcity/cullcity.cpp

```diff
@@ -85,3 +98,162 @@ void mmCullCity::Cull()
+
+        lm_dlp = GetDLPTemplate(lm_path);
+    }
+
+    BeginMemStat("mmCullCity::Init");
+    InitProblems();
+
+    Camera = camera;
+    asNode::AddChild(&BangerDataManager);
+    asNode::AddChild(&BangerActiveManager);
+    asNode::AddChild(&BangerManager);
+    asNode::AddChild(&asnode34AF0);
+    asNode::AddChild(&PHYS);
+    asNode::AddChild(&RenderWeb);
+    asNode::AddChild(&Particles);
+
+    agiMeshCardVertex* vertex = NULL;
+
+    if (MMSTATE.Weather == mmWeather::Rain)
+        vertex = RainMeshCard;
+    else
+        vertex = agiMeshSet::DefaultQuad;
+
+    Particles.Init(100, 4, 4, 4, vertex);
+
+    SnowBirthRule.SetName("SnowRule");
+    AddChild(&SnowBirthRule);
+    SnowBirthRule.Load();
+
+    RainBirthRule.SetName("RainRule");
+    AddChild(&RainBirthRule);
+    RainBirthRule.Load();
+
+    BirthRule = 0;
+
+    Sky.Init("mmsky"_xconst);
+    Loader()->EndTask(0.0f);
+
+    BeginMemStat("mmCullCity.WEB");
+    RenderWeb.Load(name, 1);
+    EndMemStat();
+
+    HitIdBound = RenderWeb.HitIdBound;
+    HitIdBound->AddRef();
+
+    BuildingChain.Init(RenderWeb.MaxCells);
+    ObjectsChain.Init(RenderWeb.MaxCells);
+    ShadowChain.Init(RenderWeb.MaxCells);
+
+    InitObjectDetail();
+
+    LastInstance = (mmInstance*) mmInstanceHeap.HeapHead;
+    ResetInst = new mmYInstance();
+
+    LoadBangers(name);
+
+    if (MMSTATE.GameMode == mmGameMode::Blitz)
+    {
+        LoadBangers(arts_formatf<64>("%s_b%d", name, MMSTATE.EventId));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::Checkpoint)
+    {
+        LoadBangers(arts_formatf<64>("%s_r%d", name, MMSTATE.EventId));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::Circuit)
+    {
+        LoadBangers(arts_formatf<64>("%s_c%d", name, MMSTATE.EventId));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::Cruise)
+    {
+        LoadBangers(arts_formatf<64>("%s_roam", name));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::CnR)
+    {
+        LoadBangers(arts_formatf<64>("%s_cops", name));
+    }
+
+    ResetInst = new mmYInstance();
+
+    LoadFacades(name);
+
+    if (CHICAGO)
+    {
+        Vector3 pos_1 = {995.782, 0.173, 1188.804};
+        Vector3 pos_2 = {995.782, 0.173, 742.411};
+        Vector3 pos_3 = {1015.383, 0.173, 1188.805};
+        Vector3 pos_4 = {1015.383, 0.173, 742.411};
+
+        mmRunwayLight* light_1 = new mmRunwayLight("fxltglow"_xconst, pos_1, pos_2);
+        mmRunwayLight* light_2 = new mmRunwayLight("fxltglow"_xconst, pos_3, pos_4);
+        BuildingChain.Parent(light_1, 35);
+        BuildingChain.Parent(light_2, 35);
+
+        InitTimeOfDayAndWeather();
+
+        if (IsSnowing)
+            InitSnowTextures();
+
+        BangerMgr()->Init(40);
+
+        if (city_dlp && city_dlp->Release())
+            Errorf("Someone is still holding a ref on the city's template");
+
+        if (lm_dlp)
+            lm_dlp->Release();
+
+        EndMemStat();
+
+        void ShowRenderStats();
```

Comment:
This is missing the CULLMGR->AddPage call.

Since `ShowRenderStats` is a static function, it wasn't automatically imported. You need to do `ARTS_IMPORT void ShowRenderStats();` outside the function, and add `import_symbol(0x48CC30, "?ShowRenderStats@@YAXXZ");` to symbols.cpp

Reply:
Added the function and the import.
`CULLMGR->AddPage(ShowRenderStats);` correctly adds the page in the game